silkworm/db/datastore/snapshots/segment/seg/decompressor.cpp

```diff
@@ -862,14 +862,15 @@ bool Decompressor::Iterator::is_next_word_compressed() const {
 Decompressor::Iterator& Decompressor::Iterator::operator++() {
     if (has_next()) {
         current_word_offset_ = word_offset_;
-        current_word_.clear();
 
         bool is_next_word_compressed = this->is_next_word_compressed();
         is_next_value_ = !is_next_value_;
         if (is_next_word_compressed) {
-            next_compressed(current_word_);
+            current_word_ = Bytes{};
```

Comment:
```suggestion
            std::get<Bytes>(current_word_).clear();
```
The idea taken from the erigon code was to not reallocate current_word_ unless it grows bigger. For files that have word size limited, this can result in a better perf.

Doing .clear() instead of reassign should work for .seg and .v files, and even .kv files if compression mode = both.

If we want to keep this optimization for .kv files with only keys or values compressed, an option could be to use a custom `union { Bytes, ByteView, flag }` instead of `std::variant`, and not touch Bytes between iterations.

Reply:
We need to check whether `current_word_` contains `Bytes` or `ByteView`, so it must be:
```
if (current_word_.holds_bytes()) {
    std::get<Bytes>(current_word_).clear();
} else {
    current_word_ = Bytes{};
}
```
I'd like to measure this change, so I'm gonna leave it for a follow-up PR.